### PR TITLE
Adds SQL Starter code sample

### LINKED ISSUE #146: Add more examples to spring-cloud-gcp-examples
- [x] Trace
- [x] CloudSQL
- [ ] Pub/Sub
- [x] Spring Integration w/ Pub/Sub

### PR DESCRIPTION
**Pull request by joaoandremartins**
Helps fix #146

### CODE REVIEW

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+package com.example;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java]: Don't know "why?", but Travis doesn't like this:
```
/home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java:24: error: Wrong order for 'java.util.List' import.
```

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/pom.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+	<dependencies>
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-gcp-starter-sql</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+			<version>1.5.4.RELEASE</version>
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/pom.xml]: Do we really need this version here?

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		this.jdbcTemplate.execute("CREATE TABLE users ("
+				+ "id int NOT NULL AUTO_INCREMENT,"
+				+ "email VARCHAR(255),"
+				+ "first_name VARCHAR(255),"
+				+ "last_name VARCHAR(255),"
+				+ "PRIMARY KEY (id));");
+		this.jdbcTemplate.execute("INSERT INTO users (email, first_name, last_name) VALUES "
+				+ "(\"luisao@gmail.com\", \"Anderson\", \"Silva\"),"
+				+ "(\"jonas@gmail.com\", \"Jonas\", \"Gonçalves\"),"
+				+ "(\"fejsa@gmail.com\", \"Ljubomir\", \"Fejsa\")");
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java]: M-m-m. Can't we rely on the DB population feature in Spring Boot: https://docs.spring.io/spring-boot/docs/1.5.7.RELEASE/reference/htmlsingle/#howto-initialize-a-database-using-spring-jdbc ?

And there won't be any questions when someone will start from the `/getTuples` request when there is nothing in DB

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+spring.cloud.gcp.sql.database-name=[database-name]
+spring.cloud.gcp.sql.instance-connection-name=[instance-connection-name]
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties]: Don't we need all other properties for `projectId`, `credentials` etc?

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java (hunk 1 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java]: I'd forgotten to organize imports before..

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java (hunk 3 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/java/com/example/WebController.java]: Yes, I was looking for that JDBC initialization but couldn't find it. Thanks for the tip!

The way it is now, I think there will always be something in the database. If the database/table doesn't exist, an error will be thrown, which might be fine for a code sample?

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties (hunk 4 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties]: Not if users are authenticated with the Cloud SDK. I added a bullet point in the README for that matter:

4. https://cloud.google.com/sdk/gcloud/reference/auth/login[Authenticate in the Cloud SDK] or add
a `spring.cloud.gcp.sql.credentials-location` property with the path on your local file system to
a credentials file, prepended with `file:`.

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+password as the value.
+
+4. https://cloud.google.com/sdk/gcloud/reference/auth/login[Authenticate in the Cloud SDK] or add
+a `spring.cloud.gcp.sql.credentials-location` property with the path on your local file system to
+a credentials file, prepended with `file:`.
+
+5. Run the `SqlApplication` Spring Boot app.
+
+6. Navigate to http://localhost:8080/populateDatabase on your browser to create a table and add
+content to it.
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc]: This bullet is obsolete already

**Comment by joaoandremartins** [comment]
Can't do changes during merge since master is protected. Please re-approve.

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+= Setup & Configuration
+
+1. Follow https://cloud.google.com/sql/docs/mysql/quickstart[these instructions] to set up a Google
+Cloud Project with billing enabled and a Google Cloud SQL instance.
+
+2. Still within the Google Cloud Console, create a new database in the instance from the
+"Databases" section.
+
+3. Open the link:src/main/resources/application.properties[application.properties] file and replace
+*[database-name]* by the name of the database you created on 2 and *[instance-connection-name]* by
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc]: nit: by -> with

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+the instance connection name of the instance you created on 1.
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc]: nit: "on 1" -> "in step 1"

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+password as the value.
+
+4. https://cloud.google.com/sdk/gcloud/reference/auth/login[Authenticate in the Cloud SDK] or add
+a `spring.cloud.gcp.sql.credentials-location` property with the path on your local file system to
+a credentials file, prepended with `file:`.
+
+5. Run the `SqlApplication` Spring Boot app. The database will be populated based on the
+link:src/main/resources/schema.sql[schema.sql] and link:src/main/resources/data.sql[data.sql] files.
+
+6. Navigate to http://localhost:8080/getTuples on your browser to print the contents of the table
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/README.adoc]: nit: on -> in

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,5 @@
+spring.cloud.gcp.sql.database-name=example
+spring.cloud.gcp.sql.instance-connection-name=sodium-gateway-790:us-central1:test
+
+# So app starts despite "table already exists" errors.
+spring.datasource.continue-on-error=true
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties]: Maybe also add `#spring.cloud.gcp.sql.credentials-location=file:/path/to/service-account.json` and `#spring.cloud.gcp.sql.password=secret` commented out for convenience.

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1,5 @@
+spring.cloud.gcp.sql.database-name=example
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties]: example -> [database-name], or just instruct the users to create a database called "example"

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties [hunk 11]
```diff
@@ -0,0 +1,5 @@
...
+spring.cloud.gcp.sql.instance-connection-name=sodium-gateway-790:us-central1:test
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/application.properties]: Forgot to remove your project? :)
`[instance-name]`

File: spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/data.sql [hunk 12]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO users VALUES
+  ("luisao@gmail.com", "Anderson", "Silva"),
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-sql-example/src/main/resources/data.sql]: These names and emails seem t0o realistic IMO.
Maybe at least change emails from `@gmail.com` to `@example.com`.

**Pull request merged**